File: src/main/java/krak/miche/KBot/commands/TestCommand.java
package krak.miche.KBot.commands;


import krak.miche.KBot.BuildVars;
import krak.miche.KBot.database.DatabaseManager;
import org.telegram.telegrambots.api.methods.send.SendMessage;
import org.telegram.telegrambots.api.objects.Chat;
import org.telegram.telegrambots.api.objects.User;
import org.telegram.telegrambots.bots.AbsSender;
import org.telegram.telegrambots.bots.commandbot.commands.BotCommand;
import org.telegram.telegrambots.exceptions.TelegramApiException;
import org.telegram.telegrambots.logging.BotLogger;


import java.sql.SQLException;

/**
 * @author Kraktun
 * @version 1.0
 * Test class for different portions of code
 */

public class TestCommand extends BotCommand {

    private static final String LOGTAG = "TESTCOMMAND";

    public TestCommand() {
        super( "test", "This command is used to test" );
    }

    @Override
    public void execute(AbsSender absSender, User user, Chat chat, String[] arguments) {
        Integer userID = user.getId();
        StringBuilder messageTextBuilder;
        if (chat.isUserChat())
        {
            if (BuildVars.SUPER_ADMINS.contains(userID))
            {
                DatabaseManager databaseManager = DatabaseManager.getInstance();
                messageTextBuilder = new StringBuilder("Send in image");
                try {
                    databaseManager.pushCommand(userID, "test");
                } catch (SQLException e) {
                    BotLogger.error(LOGTAG, e);
                }
                SendMessage answer = new SendMessage();
                answer.setChatId(chat.getId().toString());
                answer.setText(messageTextBuilder.toString());
                try {
                    absSender.execute(answer);
                } catch (TelegramApiException e) {
                    BotLogger.error(LOGTAG, e);
                }
            }
        }
    }
}

File: src/main/kotlin/com/miche/krak/kBot/commands/TestCommand.kt
package com.miche.krak.kBot.commands

import com.miche.krak.kBot.commands.core.BaseCommand
import com.miche.krak.kBot.commands.core.CommandInterface
import com.miche.krak.kBot.objects.Status
import com.miche.krak.kBot.objects.Target
import org.telegram.telegrambots.meta.api.objects.Chat
import org.telegram.telegrambots.meta.api.objects.Message
import org.telegram.telegrambots.meta.api.objects.User
import org.telegram.telegrambots.meta.bots.AbsSender

class TestCommand : CommandInterface {

    val engine = BaseCommand(
        command = "/test",
        description = "Command to test new commands",
        targets = listOf(Pair(Target.USER, Status.ADMIN)),
        exe = this
    )

    override fun execute(absSender: AbsSender, user: User, chat: Chat, arguments: List<String>, message: Message) {
    }
}